### DISCUSSION TITLE
PDF copy some seal image disappear

### DISCUSSION THREAD

**Discussion by hbh112233abc** [pymupdf/PyMuPDF#1238] [Looking for help] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
when I copy some pdf use issue [722](https://github.com/pymupdf/PyMuPDF/issues/722) demo function, and the output pdf not found some seal images.

## To Reproduce (mandatory)
the old pdf:
[55CCF322DD7C4755894B85CDD238A9B5_37.pdf](https://github.com/pymupdf/PyMuPDF/files/7074453/55CCF322DD7C4755894B85CDD238A9B5_37.pdf)
output pdf:
[55CCF322DD7C4755894B85CDD238A9B5_37_copy.pdf](https://github.com/pymupdf/PyMuPDF/files/7074456/55CCF322DD7C4755894B85CDD238A9B5_37_copy.pdf)

the right side two red seal images disappear 
![image](https://user-images.githubusercontent.com/7788164/131291091-3acd8c48-0142-49e3-8a59-d6a70c4af233.png)

demo code:
```
import fitz

src = fitz.open("old.pdf")
spage = src[0]
oldrot = spage.rotation
spage.setRotation(0)
doc = fitz.open()
npage = doc.newPage(width=spage.rect.width, height=spage.rect.height)
npage.showPDFpage(
    npage.rect,
    src,
    0,
)
npage.setRotation(oldrot)
doc.save("new.pdf")

```

## Expected behavior (optional)
the red images is pdf sign seal image, I hope keep the images at the same location,the sign info remove is ok, I don't care.

## Screenshots (optional)
old screenshot:
![image](https://user-images.githubusercontent.com/7788164/131291520-dddcabc1-2041-4698-847b-6424f00c7734.png)

new screenshot:
![image](https://user-images.githubusercontent.com/7788164/131291560-b09a2698-bfc5-40cb-800d-0f671f3c0170.png)

not found the red seal images at the right side. 

## Your configuration (mandatory)
 windows 10 64bit
python 3.7 32bit
pymupdf 1.18.17

**Comment by JorjMcKie** [comment, marked as answer]
Thank you for submitting this. This is not a bug - it works as designed.

1. As documented, `show_pdf_page()` **_cannot copy annotations of the source page._**
2. Form fields / widgets (including signatures) are a subtype of annotations, so they **_will not be included_** in the target page. The same is true for any other type of annotations - including links.

There is a solution for your case however:
You can convert the source PDF to a new PDF. The underlying MuPDF function converts annotations to normal page content. 
Then use the resulting PDF as your source.
You should now see annotation images in the target, but they technically are no longer annotations (or widgets, signatures, etc.).
Here is a snippet for how to achieve this:
```python
import fitz
import sys

src = fitz.open(sys.argv[1])  # input file
srcbytes = src.convert_to_pdf()  # convert to a new PDF
src.close()
src = fitz.open("pdf", srcbytes)  # open converted as PDF
spage = src[0]
oldrot = spage.rotation
spage.set_rotation(0)
doc = fitz.open()
npage = doc.new_page(width=spage.rect.width, height=spage.rect.height)
npage.show_pdf_page(
    npage.rect,
    src,
    0,
)
npage.set_rotation(oldrot)
doc.ez_save(__file__.replace(".py", ".pdf"))
```

**Reply by hbh112233abc** [reply]
good job! Thank you very much, your code is work, it's very good solution 👍👍👍
